.github/workflows/kind-e2e.yaml

```diff
@@ -7,92 +7,195 @@ on:
 defaults:
   run:
     shell: bash
-    working-directory: ./src/knative.dev/serving
+
+env:
+  # https://github.com/google/go-containerregistry/pull/125 allows insecure registry for
+  # '*.local' hostnames. This works both for `ko` and our own tag-to-digest resolution logic,
+  # thus allowing us to test without bypassing tag-to-digest resolution.
+  CLUSTER_DOMAIN: c${{ github.run_id }}.local
+  REGISTRY_NAME: registry.local
+  REGISTRY_PORT: 5000
+  KO_DOCKER_REPO: registry.local:5000/knative
+  KO_VERSION: 0.11.2
+  KIND_VERSION: 0.12.0
+  GOTESTSUM_VERSION: 1.7.0
+  KAPP_VERSION: 0.46.0
+  YTT_VERSION: 0.40.1
+  KO_FLAGS: --platform=linux/amd64
 
 jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Set up Go 1.17.x
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.17.x
+
+    - name: Setup Cache Directories
+      run: |
+        mkdir -p ~/artifacts/build
+        mkdir -p ~/artifacts/registry
+
+    - uses: actions/cache@v3
+      with:
+        path: |
+          ~/.cache/go-build
+           ~/go/pkg/mod
+        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-go-
+
+    - name: Install Dependencies
+      run: |
+        set -x
+        echo "::group:: install ko ${KO_VERSION}"
+        curl -L https://github.com/google/ko/releases/download/v${KO_VERSION}/ko_${KO_VERSION}_linux_x86_64.tar.gz  | tar xzf - ko
+        chmod +x ./ko
+        sudo mv ko /usr/local/bin
+        echo "::endgroup::"
+
+    - name: Setup Registry
+      run: |
+        docker run -d --restart=always \
+          -p $REGISTRY_PORT:$REGISTRY_PORT \
+          -v ~/artifacts/registry:/var/lib/registry \
+          --name $REGISTRY_NAME registry:2
 
-  e2e-tests:
-    name: e2e tests
+        # Make the $REGISTRY_NAME -> 127.0.0.1, to tell `ko` to publish to
+        # local reigstry, even when pushing $REGISTRY_NAME:$REGISTRY_PORT/some/image
+        sudo echo "127.0.0.1 $REGISTRY_NAME" | sudo tee -a /etc/hosts
+
+    - name: Build Knative
+      run: |
+        export YAML_OUTPUT_DIR=$HOME/artifacts/build
```

Comment:
this is an env var that's recognized by generate-yaml 

I've been meaning to change the inputs of the script - but it's used by prow jobs so requires some coordination